e2e/scientists_queries.yaml

```diff
@@ -403,7 +403,7 @@ queries:
         ?x ql:has-predicate ?predicate .
       }
       GROUP BY ?predicate
-      HAVING (?predicate < <Z) (?predicate = <Religion>)
+      HAVING (?predicate < <Z-Cars>) (?predicate = <Religion>)
```

Comment:
I think this is actually issue #249 and specifically the need to convert the predicate with `str()` to allow URLs to be treated as strings.

Reply:
@niklas88 If we implement the filters the way #249 suggests (Using the same schema for all types of filters) we will run into a problem with the way our vocabulary is stored. In the following example we cannot return the correct value with a single greater than filter:
```SPARQL
SELECT ?a WHERE {
  VALUES ?a {<cake3> <cake>}
  FILTER(str(?a) > "cake")
}
```
The string `<cake3>` is lexicographically smaller than `<cake>` but we want to return `<cake3>` but not `<cake>`, as `"cake3" > "cake"`

Reply:
@niklas88 I did a quick and dirty fix of the problem by treating fitlers on literals that contain either `<` or `"` differently. While that is not standard conform the proper fix to this seems to be a more significant change that might require changes to the way the vocabulary is stored and should probably be done in its own pr.